src/main/kotlin/DataStructures/Trees/BinaryTree.kt

```diff
@@ -0,0 +1,223 @@
+package DataStructures.Trees
+
+/**
+ * This entire class is used to build a Binary Tree data structure.
+ * There is the Node Class and the Tree Class, both explained below.
+ * The functions are mostly recursively implemented.
+ */
+
+
+/**
+ * A binary tree is a data structure in which an element
+ * has two successors(children). The left child is usually
+ * smaller than the parent, and the right child is usually
+ * bigger.
+ *
+ * A new Binary Tree can initialized with his root
+ */
+class BinaryTree(rootValue: Int) {
+
+    var root: Node
+    var outputBuffer: String
```

Comment:
Better not to keep this as a property here and instead pass a `StringBuilder` to the traversal methods.